### PR TITLE
Make publish site url configurable

### PR DESCRIPTION
**Pull request by timabell**
### Context

All the environments currently link to prod which is asking for mistakes
to be made.

### Changes proposed in this pull request

* config for links to publish site

#### before

Selection_676

#### after

Selection_675

### Guidance to review

:ship:

### CODE REVIEW

**Comment by timabell** [comment]
force-pushed addition of test fixup

File: config/settings/development.yml [hunk 1]
```diff
@@ -1,3 +1,5 @@
 manage_backend:
   base_url: http://localhost:3001
   secret: secret
+publish:
```
Comment (dankmitchell) [inline config/settings/development.yml]:
```suggestion
manage_frontend:
```

File: config/settings/new_qa.yml [hunk 2]
```diff
@@ -1,2 +1,4 @@
 manage_backend:
   base_url: https://api2.dev.publish-teacher-training-courses.service.gov.uk/
+publish:
```
Comment (dankmitchell) [inline config/settings/new_qa.yml]:
```suggestion
manage_frontend:
```

File: config/settings/new_staging.yml [hunk 3]
```diff
@@ -1,2 +1,4 @@
 manage_backend:
   base_url: https://api2.staging.publish-teacher-training-courses.service.gov.uk/
+publish:
```
Comment (dankmitchell) [inline config/settings/new_staging.yml]:
```suggestion
manage_frontend:
```

File: config/settings/qa.yml [hunk 4]
```diff
@@ -1,2 +1,4 @@
 manage_backend:
   base_url: https://bat-qa-mcbe-as.azurewebsites.net/
+publish:
```
Comment (dankmitchell) [inline config/settings/qa.yml]:
```suggestion
manage_frontend:
```

File: spec/features/organisations_spec.rb [hunk 5]
```diff
@@ -3,6 +3,10 @@
 RSpec.describe "Organisations", type: :feature do
   include_context 'when authenticated'
 
+  before do
+    allow(Settings.publish).to(receive(:base_url)).and_return("https://example.org")
```
Comment (dankmitchell) [inline spec/features/organisations_spec.rb]:
```suggestion
    allow(Settings.manage_frontend).to(receive(:base_url)).and_return("https://example.org")
```

File: config/settings/staging.yml [hunk 6]
```diff
@@ -1,2 +1,4 @@
 manage_backend:
   base_url: https://bat-staging-manage-courses-backend-app.azurewebsites.net/
+publish:
```
Comment (dankmitchell) [inline config/settings/staging.yml]:
```suggestion
manage_frontend:
```

File: config/settings/production.yml [hunk 7]
```diff
@@ -1,2 +1,4 @@
 manage_backend:
   base_url: https://api2.publish-teacher-training-courses.service.gov.uk
+publish:
```
Comment (dankmitchell) [inline config/settings/production.yml]:
```suggestion
manage_frontend:
```

File: app/helpers/organisation_helper.rb [hunk 8]
```diff
@@ -14,6 +14,6 @@ def provider_details(provider)
   end
 
   def provider_url_on_publish_teacher_training_courses(provider)
-    "https://publish-teacher-training-courses.education.gov.uk/organisation/#{provider.provider_code.downcase}"
+    "#{Settings.publish.base_url}/organisation/#{provider.provider_code.downcase}"
```
Comment (dankmitchell) [inline app/helpers/organisation_helper.rb]:
```suggestion
    "#{Settings.manage_frontend.base_url}/organisation/#{provider.provider_code.downcase}"
```

**Comment by timabell** [comment]
[use per-environment publish urls in support console](https://trello.com/c/HDqZ6Nxb/1708-use-per-environment-publish-urls-in-support-console)

**Comment by timabell** [comment]
Cheers for the improved name @dankmitchell

**Pull request merged**